sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -104,6 +105,31 @@ abstract class JdbcDialect extends Serializable with Logging {
    */
   def getJDBCType(dt: DataType): Option[JdbcType] = None
 
+  /**
+   * Convert java.sql.Timestamp to a LocalDateTime representing the same wall-clock time as the
+   * value stored in a remote database.
+   * JDBC dialects should override this function to provide implementations that suite their
+   * JDBC drivers.
+   * @param t Timestamp returned from JDBC driver getTimestamp method.
+   * @return A LocalDateTime representing the same wall clock time as the timestamp in database.
+   */
+  @Since("3.5.0")
+  def convertJavaTimestampToTimestampNTZ(t: Timestamp): LocalDateTime = {
+    DateTimeUtils.microsToLocalDateTime(DateTimeUtils.fromJavaTimestampNoRebase(t))
```

Comment:
The default dialect seems call `DateTimeUtils.microsToLocalDateTime` first and then call `DateTimeUtils.localDateTimeToMicros`. We should avoid this invalid conversation.

Reply:
How about `DateTimeUtils.fromJavaTimestampNoRebase(t)` only ?

Reply:
The default dialect call `DateTimeUtils.microsToLocalDateTime` first and then call `DateTimeUtils.localDateTimeToMicros`.

Reply:
Yeah. The goal is good, but we should improve the code.

Reply:
I think I am missing something here.

"`DateTimeUtils.fromJavaTimestampNoRebase(t)` only" would result in a Long value, where we wish for a `LocalDateTime` as the return type.

Reply:
For `convertTimestampNTZToJavaTimestamp`, Spark stores timestamp as long internally, then `df.collect` converts long to `LocalDateTime`, then dialect converts `LocalDateTime` to long again and creates `Timestamp`.

If we don't want to change it, we should not change `convertJavaTimestampToTimestampNTZ` either to make them  symmetric.